src/library/baseplaylistfeature.cpp

```diff
@@ -455,27 +463,34 @@ void BasePlaylistFeature::slotExportPlaylist() {
     qDebug() << "Export playlist" << playlistName;
 
     QString lastPlaylistDirectory = m_pConfig->getValue(
-            ConfigKey("[Library]", "LastImportExportPlaylistDirectory"),
+            kConfigKeyLastImportExportPlaylistDirectory,
             QStandardPaths::writableLocation(QStandardPaths::MusicLocation));
 
     // Open a dialog to let the user choose the file location for playlist export.
     // The location is set to the last used directory for import/export and the file
     // name to the playlist name.
-    QString filefilter = tr("M3U Playlist (*.m3u)");
-    QString file_location = QFileDialog::getSaveFileName(
+    QString fileFilter = tr("M3U Playlist (*.m3u)");
+    QString fileLocation = QFileDialog::getSaveFileName(
             nullptr,
             tr("Export Playlist"),
             lastPlaylistDirectory.append("/").append(playlistName),
             tr("M3U Playlist (*.m3u);;M3U8 Playlist (*.m3u8);;"
                "PLS Playlist (*.pls);;Text CSV (*.csv);;Readable Text (*.txt)"),
-            &filefilter);
+            &fileFilter);
     // Exit method if user cancelled the open dialog.
-    if (file_location.isNull() || file_location.isEmpty()) {
+    if (fileLocation.isNull() || fileLocation.isEmpty()) {
         return;
     }
-    QFileInfo fileName(file_location);
+    // If selected the extension was not added add it manually
```

Comment:
We should extract this code with the workaround and the reference to QTBUG-27186 into a utility functions instead of repeating it multiple times.